crates/forge/tests/it/zk/mod.rs

```diff
@@ -2,7 +2,40 @@
 mod basic;
 mod cheats;
 mod contracts;
+mod factory;
 mod fuzz;
 mod invariant;
 mod logs;
 mod repros;
+
+macro_rules! test_zk {
```

Comment:
Not sure this saves much typing in relation to the extra indirection it adds.

Reply:
I'm also unsure about it to be honest. In `test_repro` it kinda makes sense as there's mostly always a single input, i.e. the issue number thereby justifying the `test_repro!(4402)`. In our case we'd need to still think about a proper test name, the contract name, the test function which diminishes the need for `test_zk!(my_test_name, "Foo", "testFoo|testFooBar")`

Additionally some tests rely on checking the output where this pattern will not work. As there isn't make much utility to justify it as of now.

Reply:
You can check the result with `test_zk!(name, |runner| {...})` but you'd need to create the filter by hand - 
To be honest, most of our tests are self-contained, and we only really need to specify the test name for filtering, at least that's what I saw with our current tests.
It's rather simple macro, and the more tests we have the more it will save (I didn't refactor the other tests to use it to simplify the PR)

Reply:
It's a personal preference, but I like tests to be explicit about what they do - makes reaading tests and debugging things easier. If it turns out we have to change things often down the line, I might re-consider introducing a macro but so far it _is_ a simple macro that hides more than it helps IMO. 

If it helps others, we can keep it but I'd avoid macros unless we _really_ need them.

/2cents